File: Data Structures & Algorithms/longest-increasing-subsequence/submission-1.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        if not nums:
            return 0
        n = len(nums)
        LIS = [1] * n
        for i in range(n):
            for j in range(i):
                if nums[j] < nums[i]:
                    LIS[i] = max(LIS[i], 1+LIS[j])
        return max(LIS)

File: Data Structures & Algorithms/jump-game-ii/submission-1.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int minJumps = 0, farthestReach = 0, currentEnd = 0;
        for (int i = 0; i < nums.size() - 1; ++i) {
            farthestReach = max(farthestReach, i + nums[i]);
            if (i == currentEnd) {
                minJumps++;
                currentEnd = farthestReach;
            }
        }
        return minJumps;
    }
};